### ISSUE TITLE
Go to previous/next cursor position

### ISSUE THREAD

**Issue by spyder-issues-migrator** [spyder-ide/spyder#263]
_From [mproel...@googlemail.com](https://code.google.com/u/105494983286078857315/) on 2010-08-05T07:36:15Z_

Hello,

when I have something like:

import numpy

and goto definition of numpy, the __init__.py of numpy opens at line 1 (everything fine till now) than I go for instance to line 100 and copy the content to the clipboard than I hit CTRL+ALT+LEFT (goto previous cursor position) and paste the content to my top level file. Than I want to go back to the numpy __init__.py file hitting CTRL+ALT+RIGHT but am again online 1 and not on line 100 where I copied the file content from.

I would expect that when I hit go to previous cursor position and than goto last cursor position I end up at exactly the same location

_Original issue: http://code.google.com/p/spyderlib/issues/detail?id=263_

**Comment by spyder-issues-migrator** [comment]
_From [pierre.raybaut](https://code.google.com/u/pierre.raybaut/) on 2010-08-05T06:01:25Z_

Fixed: https://code.google.com/p/spyderlib/source/detail?r=ae6a57f621ebde7c9b7b102f050ab415fb3cb342

**Status:** Fixed

**Comment by spyder-issues-migrator** [comment]
_From [mproel...@googlemail.com](https://code.google.com/u/105494983286078857315/) on 2010-08-05T23:10:38Z_

Hello,

thanks for fixing these issues that fast.

It works nearly perfect, but in one occasion I would still expect a different behaviour:
let's take my previous example with import numpy.
1. I goto numpy (CTRL+G)
2. I copy the third line and hit CTRL+ALT+LEFT
3. I come back to my previous file and CTRL+ALT+RIGHT browses me back to line 3. So far so good.
4. But pasting now my copied content to my first file, which would look like
import numpy
===== (pasted content)

now. Now I would expect, that hitting CTRL+ALT+RIGHT browses me back to line 3 of numpy.__init__.py

In other words, I would expect, that the browse positions (the places where you go when hitting CTRL+ALT+LEFT/RIGHT) are only updated when you goto definition, or goto previous/next cursor position

**Comment by spyder-issues-migrator** [comment]
_From [pierre.raybaut](https://code.google.com/u/pierre.raybaut/) on 2010-08-07T06:10:12Z_

I've just pushed one more changeset regarding this feature, I hope this is it.
